client/me/concierge/info-step.js

```diff
@@ -55,20 +55,26 @@ class InfoStep extends Component {
 		let timezone;
 		const { signupForm, timezoneValue, translate } = this.props;
 
-		if ( ! signupForm ) {
-			message = '';
+		message = '';
```

Comment:
I moved the default logic to the reducer, however I was under the impression that reducers need to act as pure functions.

Reply:
Yeah, that's a good point.

I'm of two minds on this — on one hand you're right, `moment.tz.guess()` is not a fully pure function so introducing it as a default value makes the Reducer impure. The way around this I suppose is to default it to `null` and on `componentDidMount`, dispatch an action to set the timezone to `moment.tz.guess()` if it's `null`. We have to dispatch this into state because the value in state is what's used when the API call is made.

On the other hand — in any given environment `moment.tz.guess()` should return a deterministic value, so testing shouldn't be any harder. Replaying these actions if you're in another environment/timezone might mess things up, but we realistically never do that anyways. Dispatching on `componentDidMount` is a lot of extra work and likely will result in a flash of "UTC" before the real timezone is chosen. So pragmatically, it seems like a lot of effort for little practical gain to keep the reducer strictly pure.

I'll leave the decision to you @unDemian unless @omarjackman or @southp have strong feelings one way or another.